### DISCUSSION TITLE
Improve Mutagen Sync Efficiency For M2

### DISCUSSION THREAD

**Discussion by tdorchak** [wardenenv/warden#607] [Help & Support]
The biggest hangup I am experiencing using warden is the time it takes to complete the initial synchronization. I have added the following to my .env file to help:
`WARDEN_SYNC_IGNORE=vendor/**,var/**,pub/static/**`

With this setting, I need to run all of my composer commands in `warden shell` and then run a `composer install` outside of warden so I can debug--not ideal but workable.

I have to imagine that the `generated` directory is also causing the sync process to be slow, but I cannot work around having that sync back to my local.

I'm wondering if it would be possible to have some directories sync using mutagen's `one-way-replica` mode to improve sync time.

**Comment by navarr** [comment]
I definitely agree that initial synchronization is way too slow.  I don't think one-way sync here works as a general purpose solution, though.

For example, I often insert debugging code in `vendor/magento/*` files, or test potential patches there.  I regularly want these directories to be two-way-synced.  

That said, I don't understand why mutagen initial sync takes *so* long.  I reckon an rsync would be faster at this point, so I have to wonder what's really going on - and/or if its a bug with Mutagen

**Comment by tdorchak** [comment]
@navarr good point on the vendor use case. I haven't needed to modify anything like that since i've made the change, but that is something I have done in the past from time to time.

**Comment by navarr** [comment]
A follow-up for me, specifically, I had an antivirus with real-time file protection which was slowing sync down by 15x at least.
